### DISCUSSION TITLE
Max value that can be set in a Characteristic

### DISCUSSION THREAD

**Discussion by nicolosponziello** [h2zero/NimBLE-Arduino#335] [Q&A]
Hi, in my project I have the necessity to read a long value from a characteristic provided by the ESP32. When I set the value I get the following error: NimBLECharacteristic: "Size 3877 too large, must be no bigger than 512".
Is there a way to increase this limit? What can be the maximum value of bytes?

Thank you

**Comment by h2zero** [comment]
From the Bluetooth core specification:
>The maximum length of an attribute value shall be 512 octets.

So you will need to chunk your data into multiple writes and read the values written into a larger buffer.
